src/main/java/com/serotonin/bacnet4j/npdu/ip/IpNetwork.java

```diff
@@ -150,38 +160,66 @@ public long getBytesIn() {
+        return unicastSocket;
     }
 
     @Override
     public void initialize(final Transport transport) throws Exception {
         super.initialize(transport);
 
         localBindAddress = InetAddrCache.get(localBindAddressStr, port);
+        unicastSocket = createSocket(localBindAddress);
 
+        broadcastMAC = IpNetworkUtils.toOctetString(broadcastAddressStr, port);
+        subnetMask = BACnetUtils.dottedStringToBytes(subnetMaskStr);
+
+        if (!DEFAULT_BIND_IP.equals(localBindAddressStr) && SystemUtils.IS_OS_LINUX) {
+            // If the bind address is the wildcard address (i.e. 0.0.0.0) then we will get messages to the broadcast
+            // addresses automatically. The same is true if the OS is Windows regardless of the bind address. But on
+            // Linux we need to open a socket on the broadcast address and get the broadcasts that way.
+            try {
+                InetSocketAddress broadcastAddress = InetAddrCache.get(broadcastAddressStr, port);
+                broadcastSocket = createSocket(broadcastAddress);
+            } catch (final SocketException e) {
+                unicastSocket.close();
+                throw e;
+            }
+        }
+
+        // If the bindings were successful, start the listener threads.
+        Thread unicastThread = new Thread(() -> listen(unicastSocket),
+                "BACnet4J IP socket listener for " + transport.getLocalDevice().getId());
+        unicastThread.start();
+
+        if (broadcastSocket != null) {
+            Thread broadcastThread = new Thread(() -> listen(broadcastSocket),
+                    "BACnet4J IP broadcast socket listener for " + transport.getLocalDevice().getId());
+            broadcastThread.start();
+        }
+    }
+
+    private DatagramSocket createSocket(final InetSocketAddress bindAddress) throws SocketException {
+        LOG.info("Binding to address {}", bindAddress);
+        final DatagramSocket socket;
         if (reuseAddress) {
             socket = new DatagramSocket(null);
             socket.setReuseAddress(true);
             if (!socket.getReuseAddress())
-                LOG.warn("reuseAddress was set, but not supported by the underlying platform");
-            socket.bind(localBindAddress);
+                LOG.warn("reuseAddress was set but not supported by the underlying platform");
+            socket.bind(bindAddress);
         } else
-            socket = new DatagramSocket(localBindAddress);
+            socket = new DatagramSocket(bindAddress);
         socket.setBroadcast(true);
-
-        //        broadcastAddress = new Address(broadcastIp, port, new Network(0xffff, new byte[0]));
-        broadcastMAC = IpNetworkUtils.toOctetString(broadcastAddressStr, port);
-        subnetMask = BACnetUtils.dottedStringToBytes(subnetMaskStr);
-
-        thread = new Thread(this, "BACnet4J IP socket listener for " + transport.getLocalDevice().getId());
-        thread.start();
+        return socket;
```

Comment:
Shouldn't we always set reuseAddress=true when the bindAddress is DEFAULT_BIND_IP?

Reply:
Maybe, but this isn't an issue i was trying to solve. We have the reuse option in there, so it is controllable by the operator in any case.

Reply:
When the operator tries to bind NIC1, it also binds to the default broadcast address, which was not the case before. As a result, the default bind address is no longer available if someone tries to use both NIC1 and the default address simultaneously with reuseAddress=false.